### PR TITLE
Fix #28 - _id type

### LINKED ISSUE #28: Missing SV frequencies?
I just had a chat with two users who had noticed an abundance of seemingly rare SVs on their cases, or with other words a paucity of  variants reporting as local. Is loqusdb/loqusdbapi misbehaving or have filter settings changed in MIP? On a quick look, it does seem we have some rather common variants "missing" from loqusdb, e.g. 

Screenshot 2022-03-01 at 08 54 30

### PR DESCRIPTION
**Pull request by dnil**
### This PR adds | fixes:
- issue retrieving SV vars from the API - See #28.

### How to test:
- test a local curl for a SV variant on cg-vm1
- check that scout-stage can reach SV variants

### Review:
- [ ] Code approved by
- [ ] Tests executed by

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: loqusdbapi/models.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def id_to_str(cls, value):
 
 
 class BaseVariant(BaseModel):
-    id: str = Field(alias="_id")
+    id: Optional[Any] = Field(alias="_id")
```
Comment (northwestwitch) [inline loqusdbapi/models.py:51]: What about having either a string or a Bson object in here?

Comment (dnil) [reply loqusdbapi/models.py:51]: Or actually moving it back to the `Variant` class where it was before the issue. I'll do that!

Comment (northwestwitch) [reply loqusdbapi/models.py]: Yes, it makes more sense!

**Review by northwestwitch — APPROVED** [review]
👍🏻 Let's try it on stage!

**Comment by dnil** [comment]
Hm, apparently not enough with 'Any' (nor missing). A validator it is then?!

```
Mar 01 10:42:15 cg-vm1.scilifelab.se conmon[1906892]:   File "/usr/local/lib/python3.9/site-packages/fastapi/encoders.py", line 139, in jsonable_encoder
Mar 01 10:42:15 cg-vm1.scilifelab.se conmon[1906892]:     raise ValueError(errors)
Mar 01 10:42:15 cg-vm1.scilifelab.se conmon[1906892]: ValueError: [TypeError("'ObjectId' object is not iterable"), TypeError('vars() argument must have __dict__ attribute')]
```

File: loqusdbapi/models.py [hunk 2]
```diff
@@ -65,6 +76,7 @@ class Variant(BaseVariant):
 
 
 class StructuralVariant(BaseVariant):
+    id: Optional[PydanticObjectId] = Field(alias="_id")
```
Comment (northwestwitch) [inline loqusdbapi/models.py:79]: Why doesn't it work if you have BsonObjectId in there, like this?
`id: Optional[BsonObjectId] = Field(alias="_id")`

Comment (dnil) [reply loqusdbapi/models.py]: Since that is not a Pydantic type I guess?

Comment (northwestwitch) [reply loqusdbapi/models.py]: Ah OK

Comment (dnil) [reply loqusdbapi/models.py]: I think it has to be one of the actual Pydantic field_types there. Why Any doesn't work is beyond me at the moment; something with how the ObjectId is made.

Comment (dnil) [reply loqusdbapi/models.py]: Finally... Thanks for the coaching.

Comment (northwestwitch) [reply loqusdbapi/models.py]: Perhaps it means Any of the pydantic types

**Comment by dnil** [comment]
Ok, tested on cg-vm1. BUT needs testing with more scenarios and scout interaction before merging.
```
[hiseq.clinical@cg-vm1 ~]$ systemctl --user stop loqusdbapi.target
[hiseq.clinical@cg-vm1 ~]$ systemctl --user start loqusdbapi-rd@fix_broken_svs
[hiseq.clinical@cg-vm1 ~]$  curl -X GET "http://cg-vm1.scilifelab.se:7079/svs/?chrom=X&end_chrom=X&pos=65549568&end=88119178&sv_type=DEL"
{"chrom":"X","observations":1228,"families":["choiceracer","tightdoe","briefglider","primeasp","livingpup","topghost","tidygull","caringhound","neatkodiak","fluentsnake","trueroughy","movingbengal","superbhyena","growncrane","novelfilly","awareliger","frankswan","greatosprey","activehyena","mintlocust","suitedakita","subtlestag","poeticraptor","validmoose","intentboar","normallion","vocalrodent","gamegoblin","crackkodiak","moralmidge","actualfeline","squarequail","betterturtle","loyaloryx","onechimp","robustsnipe","happyrobin","busyloon","viableparrot","socialbream","truedeer","frankloon","neatmole","honestparrot","livingmarmot","freeelk","gladurchin","caringskunk","pumpedquail"],"_id":"5d0929a66202ae53c63bfff1","end_chrom":"X","end_left":88116180,"end_right":88122180,"end_sum":108210353266,"sv_type":"DEL","length":22569593,"pos_left":65546587,"pos_right":65552587,"pos_sum":80494892915,"total":5323}
```

**Comment by dnil** [comment]
Scout(-stage) interaction works.
Screenshot 2022-03-01 at 11 45 46

**Comment by northwestwitch** [comment]
Deployed on stage:

image

Observations are present now!

image

**Pull request merged**